src/js/media/flash.js

```diff
@@ -233,15 +233,24 @@ vjs.Flash.nativeSourceHandler = {};
  * @param  {Object} source  The source object
  * @return {String}         'probably', 'maybe', or '' (empty string)
  */
-vjs.Flash.nativeSourceHandler.canHandleSource = function(source){
+vjs.Flash.nativeSourceHandler.canHandleSource = function (source) {
   var type;
 
-  if (!source.type) {
+  function guessMimeType(src) {
+    var match = src.match(/\.([^\.]+)$/);
```

Comment:
The [regex in the html5 version](https://github.com/videojs/video.js/blob/master/src/js/media/html5.js#L330) is a little more verbose. Would it be beneficial to use the same one here?